WorkloadManagementSystem/Utilities/QueueUtilities.py

```diff
@@ -0,0 +1,225 @@
+        for tagFieldName in ('Tag', 'RequiredTag'):
+          ceTags = ceDict.get(tagFieldName, [])
+          if isinstance(ceTags, basestring):
+            ceTags = fromChar(ceTags)
+          queueTags = queueDict[queueName].get(tagFieldName)
+          if queueTags and isinstance(queueTags, basestring):
+            queueTags = fromChar(queueTags)
+            queueDict[queueName][tagFieldName] = queueTags
+          if ceTags:
+            if queueTags:
+              allTags = list(set(ceTags + queueTags))
+              queueDict[queueName][tagFieldName] = allTags
+            else:
+              queueDict[queueName][tagFieldName] = ceTags
+
+        # Some parameters can be defined on the CE level and are inherited by all Queues
+        for parameter in ['MaxRAM', 'NumberOfProcessors', 'WholeNode']:
+          queueParameter = queueDict[queueName].get(parameter)
+          ceParameter = ceDict.get(parameter)
+          if ceParameter or queueParameter:
+            queueDict[queueName][parameter] = ceParameter if not queueParameter \
+                else queueParameter
+
+        # If we have a multi-core queue add MultiProcessor tag
+        if queueDict[queueName].get('NumberOfProcessors', 1) > 1:
+          queueDict[queueName].setdefault('Tag', []).append('MultiProcessor')
+
+        queueDict[queueName]['CEName'] = ce
+        queueDict[queueName]['GridCE'] = ce
+        queueDict[queueName]['CEType'] = ceDict['CEType']
+        queueDict[queueName]['GridMiddleware'] = ceDict['CEType']
+        queueDict[queueName]['QueueName'] = queue
+
+        platform = ''
+        if "Platform" in queueDict[queueName]:
+          platform = queueDict[queueName]['Platform']
+        elif "Platform" in ceDict:
+          platform = ceDict['Platform']
+        elif "OS" in ceDict:
+          architecture = ceDict.get('architecture', 'x86_64')
+          platform = '_'.join([architecture, ceDict['OS']])
+
+        queueDict[queueName]['Platform'] = platform
+        if "Platform" not in queueDict[queueName] and platform:
+          result = getDIRACPlatform(platform)
+          if result['OK']:
+            queueDict[queueName]['Platform'] = result['Value'][0]
+
+  return S_OK(queueDict)
+
+
+def matchQueue(jobJDL, queueDict, fullMatch=False):
+  """
+  Match the job description to the queue definition
+
+  :param str job: JDL job description
+  :param bool fullMatch: test matching on all the criteria
+  :param dict queueDict: queue parameters dictionary
+
+  :return: S_OK/S_ERROR, Value - result of matching, S_OK if matched or
+           S_ERROR with the reason for no match
+  """
+
+  # Check the job description validity
+  job = ClassAd(jobJDL)
+  if not job.isOK():
+    return S_ERROR('Invalid job description')
+
+  noMatchReasons = []
+
+  # Check job requirements to resource
+  # 1. CPUTime
+  cpuTime = job.getAttributeInt('CPUTime')
+  if not cpuTime:
+    cpuTime = 84600
+  if cpuTime and cpuTime > queueDict.get('CPUTime', 0.):
```

Comment:
`cpuTime` will never be falsey
```suggestion
  if cpuTime > queueDict.get('CPUTime', 0):
```